tests/test_config_file.h

```diff
@@ -0,0 +1,157 @@
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_CONFIG_FILE_H
+#define TEST_CONFIG_FILE_H
+
+#include "core/io/config_file.h"
+
+#include "tests/test_macros.h"
+
+namespace TestConfigFile {
+
+TEST_CASE("[ConfigFile] Parsing well-formatted files") {
+	ConfigFile config_file;
+	// Formatting is intentionally hand-edited to see how human-friendly the parser is.
+	const Error error = config_file.parse(R"(
+[player]
+
+name = "Unnamed Player"
+tagline="Waiting
+for
+Godot"
+
+color =Color(   0, 0.5,1, 1) ; Inline comment
+position= Vector2(
+	3,
+	4
+)
+
+[graphics]
+
+antialiasing = true
+
+; Testing comments and case-sensitivity...
+antiAliasing = false
+)");
+
+	CHECK_MESSAGE(error == OK, "The configuration file should parse successfully.");
+	CHECK_MESSAGE(
+			String(config_file.get_value("player", "name")) == "Unnamed Player",
+			"Reading `player/name` should return the expected value.");
+	CHECK_MESSAGE(
+			String(config_file.get_value("player", "tagline")) == "Waiting\nfor\nGodot",
+			"Reading `player/tagline` should return the expected value.");
+	CHECK_MESSAGE(
+			Color(config_file.get_value("player", "color")).is_equal_approx(Color(0, 0.5, 1)),
+			"Reading `player/color` should return the expected value.");
+	CHECK_MESSAGE(
+			Vector2(config_file.get_value("player", "position")).is_equal_approx(Vector2(3, 4)),
+			"Reading `player/position` should return the expected value.");
+	CHECK_MESSAGE(
+			bool(config_file.get_value("graphics", "antialiasing")) == true,
```

Comment:
`misc/scripts/file_format.sh` doesn't like redundant `== true` (though here I tend to disagree with the script).

Reply:
I think the conditional can be safely moved to a variable up above before the assertion.
```c++
const bool config_value_enabled = config_file.get_value("graphics", "antialiasing")
CHECK_MESSAGE(config_value_enabled, "Reading `graphics/antialiasing` should return the expected value.")
```
Or `antialiasing_enabled`, but may be misleading to the purpose of testing the `ConfigFile` specifically, or combine both...

Or add static check exceptions for `tests/`. 🙂